extensions/resource-deployment/src/ui/validation/validations.ts

```diff
@@ -113,21 +128,44 @@ export class RegexValidation extends Validation {
 }
 
 export abstract class Comparison extends Validation {
-	protected _target: string; // comparison object require a target so override the base optional setting.
+	protected _target: string; // comparison object requires a target so override the base optional setting.
+	protected _onTargetValidityChangedGetter: OnTargetValidityChangedGetter; // comparison object requires an onTargetValidityChangedGetter defined so override the base optional setting.
+	protected _onNewDisposableCreated: (disposable: vscode.Disposable) => void; // comparison object requires an (disposable: vscode.Disposable) => void defined so override the base optional setting.
+	protected _listenerOnTargetValidityChangeAdded = false;
 
 	get target(): string {
 		return this._target;
 	}
 
-	constructor(validation: ComparisonValidationInfo, valueGetter: ValueGetter, variableValueGetter: VariableValueGetter) {
-		super(validation, valueGetter, variableValueGetter);
+	protected onTargetValidityChanged(onTargetValidityChangedAction: (e: boolean) => Promise<void>): void {
+		const onValidityChanged = this._onTargetValidityChangedGetter(this.target);
+		this._onNewDisposableCreated(onValidityChanged(isValid => onTargetValidityChangedAction(isValid)));
+	}
+
+	constructor(validation: ComparisonValidationInfo, validationMessageUpdater: onValidation, valueGetter: ValueGetter, targetValueGetter: TargetValueGetter, onTargetValidityChangedGetter: OnTargetValidityChangedGetter, onDisposableCreated: (disposable: vscode.Disposable) => void) {
+		super(validation, validationMessageUpdater, valueGetter, targetValueGetter);
+		this._onNewDisposableCreated = onDisposableCreated;
```

Comment:
Would suggest just using the params in the constructor instead of making new vars.

`protected _onNewDisposableCreated: (disposble: vscode.Disposable) => void` for example

Reply:
Ping

Reply:
done